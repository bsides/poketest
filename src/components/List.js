import useSWR, { mutate } from 'swr'
import fetcher from '../utils/fetcher'
import Details from './Details'

import { makeStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
})

const List = ({ url, nextUrl, previousUrl }) => {
  const { data: pokemonsReq, error } = useSWR(url, fetcher)
  const classes = useStyles()

  if (error) return <p>Falhou ao tentar acessar a API.</p>
  if (!pokemonsReq) return <p>Carregando lista de pokemons...</p>

  const { results: pokemons, next, previous } = pokemonsReq.data
  if (!pokemons) return <p>Sem dados!</p>

  // PS: this is a callback to update pagination and will
  // trigger a warning from debugger. It's ok to ignore the
  // warning as it's an intended action.
  nextUrl(next)
  previousUrl(previous)

  return (
    <>
      <TableContainer component={Paper}>
        <Table
          stickyHeader
          aria-label="table of pokemons"
          className={classes.table}
        >
          <TableHead>
            <TableRow>
              <TableCell>Sprite</TableCell>
              <TableCell>#</TableCell>
              <TableCell>Pok√©mon</TableCell>
              <TableCell>Types</TableCell>
              <TableCell>Abilities</TableCell>
              <TableCell>Forms</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {pokemons.map((pokemon) => (
              <Details url={pokemon.url} key={pokemon.url} />
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  )
}

export default List
