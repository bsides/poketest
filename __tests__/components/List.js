import React from 'react'
import renderer from 'react-test-renderer'
import List from '../../src/components/Details'

describe('List.js', () => {
  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <List
          url={`https://pokeapi.co/api/v2/pokemon/?limit=&offset=`}
          nextUrl={() => ''}
          previousUrl={() => ''}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
