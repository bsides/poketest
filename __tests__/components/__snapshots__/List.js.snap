// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.js renders the component as expected 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    <p>
      Carregando detalhes do pokemon...
    </p>
  </td>
</tr>
`;
